### ISSUE TITLE
Formatting a decimal using CultureInfo nb-NB produces differing output depending on platform.

### ISSUE THREAD

**Issue by vmchokk** [dotnet/runtime#82631]
### Description

Running the sample code below on a M1 Mac and a Windows PC produces differing output.

```c#
using System.Globalization;

var nb = CultureInfo.GetCultureInfo("nb-NB");
const decimal example = 1234.56M;

Console.WriteLine(example.ToString("N2", nb));
```

Output on Mac: `1,234.56`
Output on Windows PC: `1 234,56`

It is expected that the code produces the same output regardless of platform.

### Configuration

* Which version of .NET is the code running on?
  * Tested on 7.0.102 and 6.0.405
* What OS and version, and for Linux, what distro?
  * MacOS Ventura 13.0
* What is the architecture (x64, x86, ARM, ARM64)?
  * Apple M1 Max
* Do you know whether it is specific to that configuration?
  * No

### Regression?

Tested with v6.0.405

**Comment by steveisok** [comment]
@tarekgh we aren't using custom ICU on the osx desktop.  Has that changed?

**Comment by tarekgh** [comment]
@steveisok I recall there were some changes to pick the globalization data directly from the OS and not from ICU when running on mac-os. Isn't that the case? I may be wrong :-)

**Comment by steveisok** [comment]
Detecting the default locale was changed, but this should only be a fallback on OSX when the `LANG` environment variable isn't set. OSX should have that always set - even when you switch locales.

https://github.com/dotnet/runtime/blob/29e333d41c7a3d2b571505dc780e146e8058974c/src/native/libs/System.Globalization.Native/pal_locale.c#L220-L230

Outside of that, I don't think anything else was changed recently.

**Comment by tarekgh** [comment]
Thanks @steveisok. 

@vmchokk I investigated this and found that mac OS comes with a customized ICU version exist in the path `/usr/lib/libicucore.A.dylib`. This customized version is setting the `Decimal Separator = "."` and `Group Separator = ","` for the `nb-NB` culture. 
These values are different than what Unicode standard is providing through CLDR/ICU. The provided Unicode standard values are `Decimal Separator = ","` and `Group Separator = " "`.

I tried to install the ICU headers and libraries files and created a C++ app which statically linked to the non-mac OS ICU libraries to ensure not using the mac OS built version of ICU. And I found this c++ app produces the correct standard values `Decimal Separator = ","` and `Group Separator = " "`.

To summarize, you are seeing the discrepancy when running on mac OS because the OS is customizing the values and not using the standard values. The .NET is just giving whatever the OS is providing. You may report this to Apple and try to find out the reason behind this customization. Thanks for your report. I am closing the issue as .NET is working as expected but the OS is overwriting the correct culture values.
